project_setup/scripts/iam-to-github

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python
+
+"""Create GitHub secrets from AWS credentials extracted from Terraform state.
+
+This command must be executed in the directory containing the .terraform state
+within a GitHub project.  It will attempt to detect the repository name from
+the projects git origin.
+
+It requires a Personal Access Token from GitHub that has "repo" access scope.
+
+Usage:
+  iam-to-github [options] <github-personal-access-token>
+  iam-to-github (-h | --help)
+
+Options:
+  -h --help              Show this message.
+  --log-level=LEVEL      If specified, then the log level will be set to
+                         the specified value.  Valid values are "debug", "info",
+                         "warning", "error", and "critical". [default: info]
+  --repo=REPONAME        Use provided repository name instead of detecting it.
+  --suffix=SUFFIX        Append a suffix to the secret names.
+  --user=FILTER          Only process users that contain the filter text.
+"""
+
+# Standard Python Libraries
+from base64 import b64encode
+import json
+import logging
+import re
+import subprocess  # nosec : security implications have been considered
+import sys
+from typing import Any, Dict, Generator, List, Optional, Tuple
+
+# Third-Party Libraries
+import docopt
+from nacl import encoding, public
+import requests
+from schema import And, Or, Schema, SchemaError, Use
+
+GIT_URL_RE: re.Pattern = re.compile("git@github.com:(.*).git")
```

Comment:
Since this will only work if the repository has been cloned with an SSH configuration I think we should note that somewhere.

Reply:
Good catch.  I've got a better idea!  

https://regexr.com/4uvj8

Reply:
Fixed in: ff55608

Reply:
Nice. Didn't want to ask too much x)